mewarpx/mewarpx/util.py

```diff
@@ -34,6 +50,108 @@ def init_libwarpx(ndim, rz):
     assert pywarpx._libwarpx
 
 
+def get_velocities(num_samples, T, m, emission_type='thermionic',
+                   transverse_fac=1.0, rseed=None):
+
+    """Generate array of random [vx, vy, vz] for cathode-emitted electrons.
+    Arguments:
+        num_samples (int): Number of particles to generate velocities for
+        T (float): Temperature for the electrons (usually material temp) (K)
+        m (float): Mass of elementary particle (kg)
+        emission_type (str): Use "thermionic" for a thermionic emitter oriented
+            along +zhat, and use "random" for a purely thermal distribution
+            with no preferred direction. "half_maxwellian" is used at present
+            for surface ionization, again along +zhat. Defaults to
+            "thermionic".
+        transverse_fac (float): Scale the particle's x and y average energies
+            by this factor, scales z average energy to conserve total average
+            particle energy in the distribution. Default 1., Min 0., Max 2.
+        rseed (positive int): If specified, seed the random number generator.
+            Used for testing. The random number generator is set back at the
+            end of the function.
+    Returns:
+        velocities (np.ndarray): array of shape (num_samples, 3) with (vx, vy,
+        vz) for each electron.
+    """
+
+    if (emission_type != 'thermionic') and not np.isclose(transverse_fac, 1.0):
+        return ValueError('transverse_fac is a support argument only for '
+                          + 'thermionic emissiion models!')
+
+    if rseed is not None:
+        nprstate = np.random.get_state()
+        np.random.seed(rseed)
+    sigma = np.sqrt(constants.kb_J * T / m)
+
+    if transverse_fac < 0.:
+        print("WARNING: transverse_fac is out of bounds")
+        print("Constraining to minimum value of 0.")
+        beta = 0.
+    elif transverse_fac > 2.:
+        print("WARNING: transverse_fac is out of bounds")
+        print("Constraining to maximum value of 2.")
```

Comment:
Let's use `warnings` package (eg `warnings.warn` I think?) instead of these print statements.